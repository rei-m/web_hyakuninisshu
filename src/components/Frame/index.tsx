import * as React from 'react';
import styled, { StyledFunction } from 'styled-components';
import Header from '../Header';
import Navigation from '../Navigation';
import { MenuType } from '../../enums';

interface BodyProps {
  isDisplayNav: boolean;
}

const div: StyledFunction<BodyProps & React.HTMLProps<HTMLElement>> =
  styled.div;

export const Body = div`
  padding-top: 56px;
  padding-bottom: ${props => (props.isDisplayNav ? '56px' : '0')};
  text-align: center;
  background-color: #fffff0;
  min-height: 100vh;

  @media screen and (min-width: 768px) {
    padding-top: 64px;
    padding-bottom: 0;
    padding-left: ${props => (props.isDisplayNav ? '64px' : '0')};
  }
`;

export interface FrameProps {
  readonly subTitle: string;
  readonly canBack: boolean;
  readonly currentMenuType?: MenuType;
  readonly isDisplayNav: boolean;
  readonly onClickBack: () => void;
}

const Frame: React.StatelessComponent<FrameProps> = ({
  canBack,
  children,
  subTitle,
  onClickBack,
  isDisplayNav,
  currentMenuType
}) => (
  <div>
    <Header subTitle={subTitle} canBack={canBack} onClickBack={onClickBack} />
    <Body isDisplayNav={isDisplayNav}>{children}</Body>
    {isDisplayNav && <Navigation currentMenuType={currentMenuType} />}
  </div>
);

export default Frame;
