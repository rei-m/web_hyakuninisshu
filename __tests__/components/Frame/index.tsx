import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { MemoryRouter } from 'react-router-dom';
import Frame, { FrameProps } from '@src/components/Frame';

describe('<Frame />', () => {
  let baseProps: FrameProps;

  beforeEach(() => {
    baseProps = {
      canBack: false,
      description: 'This is description',
      isDisplayNav: false,
      onClickBack: jest.fn(),
      subTitle: 'This is subtitle'
    };
  });

  it('should render component when not exist navigation', () => {
    const renderer = ReactTestRenderer.create(
      <MemoryRouter>
        <Frame {...baseProps} />
      </MemoryRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when exist navigation', () => {
    const props = { ...baseProps, isDisplayNav: true };
    const renderer = ReactTestRenderer.create(
      <MemoryRouter>
        <Frame {...props} />
      </MemoryRouter>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
