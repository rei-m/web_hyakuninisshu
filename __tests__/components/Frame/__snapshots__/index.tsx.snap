// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Frame /> should render component when exist navigation 1`] = `
<div>
  <header
    className="sc-bwzfXH kabzOD"
  >
    <h1
      className="sc-htpNat iTzsdR"
    >
      百人一首 - 
      This is subtitle
       -
    </h1>
  </header>
  <div
    className="sc-gzVnrw jdRdke"
  />
  <nav
    className="sc-EHOje iVKSta"
  >
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa gNYCTH"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/training"
          onClick={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-ifAKCX hrwKKP"
          >
            create
          </i>
        </a>
      </div>
    </span>
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa gNYCTH"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/exam"
          onClick={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-ifAKCX iiZnFL"
          >
            note
          </i>
        </a>
      </div>
    </span>
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa gNYCTH"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/karutas"
          onClick={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-ifAKCX ihzbNI"
          >
            library_books
          </i>
        </a>
      </div>
    </span>
    <span
      className="sc-bZQynM jmnWoK"
    >
      <div
        className="sc-bdVaJa gNYCTH"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/about"
          onClick={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-ifAKCX jzZElc"
          >
            settings
          </i>
        </a>
      </div>
    </span>
  </nav>
</div>
`;

exports[`<Frame /> should render component when not exist navigation 1`] = `
<div>
  <header
    className="sc-bwzfXH kabzOD"
  >
    <h1
      className="sc-htpNat iTzsdR"
    >
      百人一首 - 
      This is subtitle
       -
    </h1>
  </header>
  <div
    className="sc-gzVnrw dWyeKG"
  />
</div>
`;
